composer.json

```diff
@@ -20,16 +20,12 @@
     },
     "autoload": {
         "psr-4": {
-            "Koriym\\Attributes\\": "src/"
+            "Koriym\\Attributes\\": "src"
         }
     },
     "autoload-dev": {
         "psr-4": {
-            "Koriym\\Attributes\\": [
-                "tests/",
-                "tests/Fake",
-                "tests-php8"
-            ]
+            "Koriym\\Attributes\\Tests\\": "tests"
```

Comment:
It's safe to have unique namespace, one per tests and one per src. That way classes cannot collide with each other, like in this case:

```bash
src/DuplicatedClass.php
tests/DuplicatedClass.php
```

Reply:
`Fake` prefix prevent to collides and "looks" fake clearly. And since it is the same namespace, you can reduce a lot of `use` import.

I follow this style from @pmjones 's @auraphp libraries
https://github.com/auraphp/Aura.Sql/blob/3.x/composer.json#L39

It's not a big deal ('use' statements can be folded in most editors), but it's a style we' ve been used to for years.

Reply:
> Fake prefix prevent to collides and "looks" fake clearly. And since it is the same namespace, you can reduce a lot of use import.

Understood. Actually this line itself is broken, without any extra prefixes:

```php
"Koriym\\Attributes\\": [
                "tests/",
```

> And since it is the same namespace, you can reduce a lot of use import.

That's the problematic part that opens space for bugs. In same namespace you can have `SomeService` and `SomeService` that is a test. It usually works, because PHPUnit has own autoloader, but it can also bite you hard, when the tool depends on composer.

But let me know if that's a blocker for you, I'll try to somehow revert it.

Reply:
> In same namespace you can have SomeService and SomeService 

It is technically possible, but we don't accept PRs for mock code without the `Fake` prefix, and it has never been a problem with many repositories (@auraphp @bearsunday @korym 30-50+?) for many years. (nearly 10?)

Please revert it if it's very easy. But if you needs some more work, No problem. I will try new practice ;)

Reply:
A lot of my styles I learned from @auraphp.(thanks @pmjones)  I think style is a policy that includes preference rather than superiority, and I think @auraphp prefers clean and less description code.

Reply:
I think it would save me an hour of work now if we can skip it. Renaming files now after more commits would be mostly manual work :/

I haven't used auraphp, but I suggest re-thinking approach not based on authority but rather bug-prone rate. Even Symfony uses lot of bad practises that should be avoided :)

Personally I use Occams' Razor and Boy Scout rule, so if I can prevent someone in the future from a that is technically possible and happened to me, I'll do it.

Reply:
I wrote about it here https://tomasvotruba.com/blog/2019/05/02/alias-as-a-code-smell/, when refactoring big project full of duplicate class names. Such a simple approach made the project extreemely heavy for cognition.

Reply:
> I suggest re-thinking approach not based on authority

I suppose I should be flattered to be called an authority.

>  but rather bug-prone rate.

Do you have some numbers around different practices, regarding this?

Reply:
Any organization used as resource is :)

Not really, just personal experience with legacy projects.

Reply:
> I think it would save me an hour of work now if we can skip it. Renaming files now after more commits would be mostly manual work :/

Sure. No problem.

> Even Symfony uses lot of bad practices that should be avoided :)

:) Yes. That's still possible after a lot of discussion. 

> on "authority"

Nice! (cc/ @pmjones)  :) But actually it's also based on my personal experience too. @auraphp @bearsunday @koriym or @ray-di use that style.

>  https://tomasvotruba.com/blog/2019/05/02/alias-as-a-code-smell/

Thanks for the your blog post link! But I have mixed feelings about this, and in fact I tend to intentionally use the same short names (I don't line up Queries though), but I understand your intent. 

Again, I'm OK for this. Let's go forward! Arigato.